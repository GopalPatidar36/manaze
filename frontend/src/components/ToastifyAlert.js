import React from "react";
import { ToastContainer, Slide, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

export const MESSAGE = {
  ticketCreated: "Ticket created successfully",
  ticketUpdated: "Ticket updated successfully",
  ticketDeleted: "Ticket deleted successfully",
  userCreated: "User created successfully",
  userUpdated: "User updated successfully",
};

export const alertMessage = (message, type = "success") => {
  toatMessage(message, type);
};

export const toatMessage = (message, type) => {
  if (type === "error") toast.error(message, { theme: "colored" });
  else if (type === "success") toast.success(message);
};

function ToastifyAlert() {
  return (
    <ToastContainer
      transition={Slide}
      autoClose={2000}
      hideProgressBar
      position="top-center"
      theme="light"
      closeOnClick={false}
      pauseOnHover={true}
      progress={false}
    />
  );
}

export default ToastifyAlert;
