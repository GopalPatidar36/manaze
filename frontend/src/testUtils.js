// src/testUtils.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { Provider } from "react-redux";
import { MockedProvider } from "@apollo/client/testing";
import { BrowserRouter as Router } from "react-router-dom";
import store from "./store";

function render(ui, { preloadedState, apolloMocks, ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          <Router>{children}</Router>
        </MockedProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
