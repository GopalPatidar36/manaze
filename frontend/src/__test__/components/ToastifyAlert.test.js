import ToastifyAlert from "../../components/ToastifyAlert";
import { render } from "../../testUtils";

describe("Testing with ToastifyAlert data", () => {
  let components;
  beforeEach(async () => {
    components = render(<ToastifyAlert />);
  });

  it("should match dom snapshot:-ToastifyAlert", () => {
    expect(components.asFragment()).toMatchSnapshot();
  });
});
