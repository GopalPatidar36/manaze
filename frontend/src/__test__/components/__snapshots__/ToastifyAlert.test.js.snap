// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with ToastifyAlert data should match dom snapshot:-ToastifyAlert 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
